### PR TITLE
fix: Unmuting users on average rooms allows them to speak if room is later set as read-only

### PR DESCRIPTION
**Pull request by heitortanoue**
## Proposed changes (including videos or screenshots)

Users can no longer send messages in a read only room if they are not unmuted by an admin. 
Unmuting an user doesn't add it to the `unmuted` array anymore, just removing it from the `muted` array

## Issue(s)
#15378 

## Steps to test or reproduce

Action | Effect | read-only | muted | unmuted
------------ | ------------- | ------------- | ------------- | -------------
Chat `x` has participant `random-dude` (it's newly created) | Everybody can send messages | `false` | `[]` | `[]`
`random-dude` sends spam & therefore gets muted | Everybody but the `random-dude` can send messages | `false` | `['random-dude']` | `[]`
After stopping spamming, `random-dude` gets re-activated | Everybody can send messages  | `false` | `[]` | `['random-dude']` ❗️
Now, the entire chat needs to be muted for some time -> moderator sets it to read-only |  Nobody except the `random-dude` ❗️ can send messages | `true` | `[]` | `['random-dude']`

For testing the unmuting on the read only rooms I recommend being in an admin account and using the slash command `/unmute <username>`.

## Further comments

TC-815

### CODE REVIEW

**Review by pierre-lehnen-rc — COMMENTED** [review]
Can you apply the same change to the `muteUserInRoom` method?

Otherwise if a user loses the permission to speak on a readonly room, they'll remain muted if the room stops being readonly.

**Review by gabriellsh — DISMISSED** [review]
LGTM

File: apps/meteor/server/models/raw/Rooms.ts [hunk 1]
```diff
@@ -1707,7 +1707,31 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 		return this.updateOne(query, update);
 	}
 
-	unmuteUsernameByRoomId(_id: IRoom['_id'], username: IUser['username']): Promise<UpdateResult> {
+	muteReadOnlyMutedUsernameByRoomId(_id: IRoom['_id'], username: IUser['username']): Promise<UpdateResult> {
```
Comment (MarcosSpessatto) [inline apps/meteor/server/models/raw/Rooms.ts:1710]: Would it not be better to be `muteReadOnlyUsernameByRoomId`?

File: apps/meteor/server/models/raw/Rooms.ts [hunk 2]
```diff
@@ -1707,7 +1707,31 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
...
+		const query: Filter<IRoom> = { _id };
```
Comment (MarcosSpessatto) [inline apps/meteor/server/models/raw/Rooms.ts:1711]: Doesn't it make sense to query also by readonly here just to make sure and to make the method name make more sense? `const query: Filter<IRoom> = { _id, ro: true };`

File: apps/meteor/server/models/raw/Rooms.ts [hunk 3]
```diff
@@ -1707,7 +1707,31 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
...
+		const update: UpdateFilter<IRoom> = {
+			$pull: {
+				muted: username,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	unmuteReadOnlyUsernameByRoomId(_id: string, username: string): Promise<UpdateResult> {
```
Comment (MarcosSpessatto) [inline apps/meteor/server/models/raw/Rooms.ts:1734]: I think we could also add the `{ to: true }` flag here as well, just to reflect the method name.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

**Review by matheusbsilva137 — APPROVED** [review]

**Review by MarcosSpessatto — APPROVED** [review]
It'd be nice if we could add a test to cover this scenario, We can add that in a subsequent PR.

**Review by KevLehman — APPROVED** [review]

**Pull request merged**